client/src/components/layout/ShareTimeTable.jsx

```diff
@@ -1,6 +1,232 @@
+  const isValidNewOption = (inputValue, selectValue) =>
+    inputValue.length > 0 && selectValue.length < 5;
+
+  const [userInfo, loading] = useGetData("/api/share/shareTT/");
+  return (
+    <>
+      <p className='title'>
+        Please Enter your Branch and year below:
+      </p>
+      <form className='form-whole' onSubmit={onSubmit}>
+        <div className='container-helform'>
+          <Creatable
+            components={{ Menu }}
+            onChange={handleBranchChange}
+            value={branch}
+            isMulti
+            isValidNewOption={isValidNewOption}
+            options={branch && branch.length >= 2 ? branch : branches}
+            className='left-width branch-inp'
+            placeholder='Select branch (select 2 branches for dual degree)'
+            theme={theme => ({
+              ...theme,
+              borderRadius: 2,
+              colors: {
+                ...theme.colors,
+                primary25: "#fecb6e",
+                neutral0: "#f9e3b4"
+              }
+            })}
+          />
+          <p className='label-mod branch-inp'>Select year: </p>
+          <FormControl component='fieldset' className='radio-grp'>
+            <RadioGroup
+              row
+              aria-label='position'
+              name='position'
+              defaultValue='End'
+              className='radio-grp'
+            >
+              <FormControlLabel
+                value='1'
+                control={<Radio color='primary' />}
+                label='First'
+                className='text-black'
+                onChange={handleYearChange}
+              />
+              <FormControlLabel
+                value='2'
+                control={<Radio color='primary' />}
+                label='Second'
+                className='text-black'
+                onChange={handleYearChange}
+              />
+              <FormControlLabel
+                value='3'
+                control={<Radio color='primary' />}
+                label='Third'
+                className='text-black'
+                onChange={handleYearChange}
+              />
+              <FormControlLabel
+                value='4'
+                control={<Radio color='primary' />}
+                label='Fourth'
+                className='text-black'
+                onChange={handleYearChange}
+              />
+              <FormControlLabel
+                value='5'
+                control={<Radio color='primary' />}
+                label='Fifth'
+                className='text-black'
+                onChange={handleYearChange}
+              />
+            </RadioGroup>
+          </FormControl>
+        </div>
+        <div className='form-group'>
+          <input
+            type='submit'
+            className='btn btn-primary btn-hf btn-big'
+            value='Submit'
+          />
+        </div>
+      </form>
+      {!loading && formData["TTs"].length !== 0 ? (
+        formData["TTs"].map(item => {
```

Comment:
```suggestion
      {!loading && TTs.length !== 0 ? (
        TTs.map(item => {
```

You're destructuring at the top, so you can just use that variable.

Reply:
done